merc_module/mercmodule.py

```diff
@@ -499,180 +623,196 @@ def MakeBigRand(
-                        for i in range(len(SmallFirstData))]
-        SmallFirstData=numpy.array(SmallFirstData)
-        d=str(SmallFirstData[SmallFirstData[:,0]=='Plantsml',1][0])
-        m=str(SmallFirstData[SmallFirstData[:,0]=='Plantsml',2][0])
+        SmallFirstData = [SmallFirstData[i].split() for i in range(len(SmallFirstData))]
+        SmallFirstData = numpy.array(SmallFirstData)
+        d = str(SmallFirstData[SmallFirstData[:, 0] == "Plantsml", 1][0])
+        m = str(SmallFirstData[SmallFirstData[:, 0] == "Plantsml", 2][0])
 
         ### Format data as the first line of each big.in object entry
-        SmallFirstLines=[small[i].ljust(10)+'d= '+d+'  m= '+m+'  r= 0.001\n'
-                         for i in range(len(small))]
+        SmallFirstLines = [
+            small[i].ljust(10) + "d= " + d + "  m= " + m + "  r= 0.001\n"
+            for i in range(len(small))
+        ]
 
         ### Read generic big.in file header
         SmallHeader = MercModule.FILE_CONTENTS_SMALL_HEADER
 
         ### No spin for all objects
-        smalls=["  0.0  0.0  0.0\n"] * len(small)
+        smalls = ["  0.0  0.0  0.0\n"] * len(small)
 
         ### Write data
-        MercModule.WriteObjInFile(current_dir=here,
-                                  subdirectory=whichdir,
-                                  names=small,
-                                  filename='small',
-                                  Header=SmallHeader,
-                                  FirstLines=SmallFirstLines,
-                                  xv=smallxv,
-                                  s=smalls)
+        MercModule.WriteObjInFile(
+            current_dir=here,
+            subdirectory=whichdir,
+            names=small,
+            filename="small",
+            Header=SmallHeader,
+            FirstLines=SmallFirstLines,
+            xv=smallxv,
+            s=smalls,
+        )
 
     @staticmethod
-    def Good2Small(whichdir,whichtime,n):
+    def Good2Small(whichdir, whichtime, n):
         """Copy the small rocks from good.in to small.in"""
         assert type(whichdir) is str
         assert type(whichtime) is str
         assert type(n) is int
-        assert n>=0
+        assert n >= 0
 
-        here=os.getcwd()
-        print('Good2Small '+whichdir+'/good.in  '+whichtime)
+        here = os.getcwd()
+        print("Good2Small " + whichdir + "/good.in  " + whichtime)
 
         ### Read in objects from good.in file
-        #	goodin=open(whichdir+'/good.in','r')
-        #	GoodLen=MercModule.FileLength(whichdir+'/good.in')
-        goodin=open('good.in','r')
-        GoodLen=MercModule.FileLength('good.in')
-        if (GoodLen%4 != 0):
-            print('??? good.in length = '+str(GoodLen))
-        ngood=GoodLen/4
+        # goodin=open(whichdir+'/good.in','r')
+        # GoodLen=MercModule.FileLength(whichdir+'/good.in')
+        # @TODO: Needs to be made an absolute path?  Looks like it used to use the subdirectory?
```

Comment:
Yeah, in the CopyInfo function that creates it, the path is `f"{here}/{whichdir}/good.in"` so should make sure this matches. (Ideally there'd be some class-level filename definition rather than having it separately in multiple places, I suppose?)